### ISSUE TITLE
Vaultwarden silently falls back to SQLite if DATABASE_URL is invalid

### ISSUE THREAD

**Issue by netthier** [dani-garcia/vaultwarden#2835]
### Subject of the issue

When an invalid `DATABASE_URL` is specified, e.g. with a typo such as `postgreql://user:pwd@host` or with quotes like `"postgresql://user:pwd@host"`, Vaultwarden silently falls back to using the SQLite database without logging any errors or warnings.
### Deployment environment
Used image is `vaultwarden/server:1.26.0`

### Steps to reproduce

I started vaultwarden via a docker-compose file with the `DATABASE_URL` variable set

### Expected behaviour

If an invalid `DATABASE_URL` is specified, Vaultwarden should either log an error or a warning along the lines of "Failed to parse DATABASE_URL, falling back to SQLite"
### Actual behaviour

Vaultwarden starts regularly and runs migrations against an SQLite database.

**Comment by tessus** [comment]
I've noticed the same beavior and IMO it should not fall back to sqlite, but abend with an error message.

**Comment by BlackDex** [comment]
Where did you configured this? Docker compose file? Custom export? Systemd file. And could you give an example? Without the actual password etc if course.

**Comment by netthier** [comment]
> Where did you configured this? Docker compose file? Custom export? Systemd file. And could you give an example? Without the actual password etc if course.

As mentioned in the issue, I specified this in my docker-compose file. The issue also contains examples without actual passwords. 
I noticed this issue after accidentally specifying the URL like this:
```yaml
environment:
  - DATABASE_URL="postgresql://user:pwd@host/vaultwarden"
 ```
Specifying it as `"DATABASE_URL=postgresql://user:pwd@host/vaultwarden"` instead worked.
Before opening this issue I further investigated by intentionally introducing a typo in the working variant, and removing a letter from the protocol, ending up with `postgreql://etc`, which led to the same issue.

**Comment by BlackDex** [comment]
Thanks for the extra info.
The thing is that the quotes are an issue with compose, but not with export or a .env loaded by Vaultwarden it self.

But, it indeed should not continue if there is a faulty setting.
The only setting which could be seen as faulty but still should fall back is an empty string, since that indicates a default.

Thanks for the report.

**Comment by BlackDex** [comment]
Should be resolved via #2873 already.
